app/main/views/suppliers.py

```diff
@@ -479,6 +479,42 @@ def set_a_declaration(supplier_id, framework_slug):
     return jsonify(declaration=supplier_framework.declaration), status_code
 
 
+@main.route('/suppliers/<int:supplier_id>/frameworks/<framework_slug>/declaration', methods=['POST'])
+def remove_a_declaration(framework_slug, supplier_id):
+    """
+    This route will replace unsuccessful applicants's declarations with an empty dict and return the altered object.
+    :param framework_slug: a string describing the framework
+    :type framework_slug: string
+    :param supplier_id: a way of identifying the supplier whose declaration will be removed
+    :type supplier_id: int
+    :return: The altered SupplierFramework object
+    :rtype: SupplierFramework
+    """
+    updater_json = validate_and_return_updater_request()
+    supplier_framework = SupplierFramework.find_by_supplier_and_framework(
+        supplier_id, framework_slug
+    ).first()
```

Comment:
(I think) you can use `.first_or_404()` here to give a nice behaviour when the supplier/framework combination doesn't exist (rather than a 500 which is what would happen here).

Personally, when doing this sort of operation, I would also take out a write lock on the `SupplierFramework` row by adding a use of `.with_for_update()`, specifically in this case `.with_for_update(of=SupplierFramework)`. This prevents any race conditions over the modification of the row.